synapse/handlers/user_directory.py

```diff
@@ -357,12 +352,8 @@ async def _handle_new_user(
             # First, if they're our user then we need to update for every user
             if self.is_mine_id(user_id):
```

Comment:
think this is redundant?

Reply:
Because we'll check to see if they're local within `is_excluded_from_user_dir`? I agree...

But. I think putting the localness check into `is_excluded_from_user_dir` makes things more confusing. To a first approximation, it's true that we'll never exclude a remote user from the user directory. With one exception: `users_who_share_private_rooms`. This contains triples `(local_user_id, any_user_id, room_id)` where the two users both belong to the given private room. Only local users live in the first entry of those tuples. (I think this is an optimisation?) To maintain that there's extra logic which needs to ensure we don't put remote users into that column.

I think it's going to be clearer if we have a function `should_include_local_user_in_dir` and make the local-vs-remote logic explicit.